docs/standard/native-interop/interop-guidance.md

```diff
@@ -0,0 +1,247 @@
+
+> __Windows Specific__  
+> For `[Out]` strings the CLR will use `CoTaskMemFree` by default to free strings or `SysStringFree` for strings that are marked
+as `UnmanagedType.BSTR`.  
+> **For most APIs with an output string buffer:**  
+> The passed in character count must include the null. If the returned value is less than the passed in character count the call has succeeded and the value is the number of characters *without* the trailing null. Otherwise the count is the required size of the buffer *including* the null character.  
+> - Pass in 5, get 4: The string is 4 characters long with a trailing null.
+> - Pass in 5, get 6: The string is 5 characters long, need a 6 character buffer to hold the null.  
+> [Windows Data Types for Strings](/windows/desktop/Intl/windows-data-types-for-strings)
+
+## Boolean parameters and fields
+
+Booleans are easy to mess up. The default marshalling for .NET is as the Windows type `BOOL`, where it is a 4 byte value. However, the `_Bool`, and `bool` types in C and C++ are a *single* byte. This can lead to hard to track down bugs as half the return value will be discarded, which will only *potentially* change the result. See the documentation on [customizing boolean field marshalling](customize-struct-marshalling#customizing-boolean-field-marshalling) for information on marshalling .NET `bool` values to C or C++ `bool` types.
+
+## Guids
+
+Guids are usable directly in signatures. Many Windows APIs take `GUID&` type aliases like `REFIID`. When passed by ref they can either be passed by `ref` or with the `[MarshalAs(UnmanagedType.LPStruct)]` attribute.
+
+| Guid | By-ref Guid |
+|------|-------------|
+| `KNOWNFOLDERID` | `REFKNOWNFOLDERID` |
+
+**❌ DO NOT** Use `[MarshalAs(UnmanagedType.LPStruct)]` for anything other than `ref` Guid parameters.
+
+## Blittable Types
+
+Blittable types are types that have the same bit-level representation in managed and native code. As such they do not need to be converted to another format to be marshalled to and from native code, and as this improves performance they should be preferred.
+
+**Blittable types:**
+
+- `byte`, `sbyte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `single`, `double`
+- non-nested one dimensional arrays of blittable types (e.g. `int[]`)
+- structs and classes with fixed layout that only have blittable value types for instance fields
+  - fixed layout requires `[StructLayout(LayoutKind.Sequential)]` or `[StructLayout(LayoutKind.Explicit)]`
+  - structs are `LayoutKind.Sequential` by default, classes are `LayoutKind.Auto`
+
+**NOT blittable:**
+
+- `bool`
+
+**SOMETIMES blittable:**
+
+- `char`, `string`
+
+When blittable types are passed by reference they are simply pinned by the marshaller instead of being copied to an intermediate buffer. (Classes are inherently passed by reference, structs are passed by reference when used with `ref` or `out`.)
```

Comment:
```suggestion
When blittable types are passed by reference, they're simply pinned by the marshaller instead of being copied to an intermediate buffer. (Classes are inherently passed by reference, structs are passed by reference when used with `ref` or `out`.)
```